lib/statsd.js

```diff
@@ -40,24 +41,27 @@ var Client = function (host, port, prefix, suffix, globalize, cacheDns, mock, gl
   this.port        = options.port || 8125;
   this.prefix      = options.prefix || '';
   this.suffix      = options.suffix || '';
-  this.socket      = dgram.createSocket('udp4');
+  this.socket      = options.isChild ? options.socket : dgram.createSocket('udp4');
```

Comment:
Can you add a comment to the ones you are overriding as to why they need to be overriden for the child case?  I think I get why for each of them, but a comment would help folks first looking at this as to why only some get this treatment.